zeus/carbon/carbon_intensity_provider.py

```diff
@@ -0,0 +1,19 @@
+"""Abstract Carbon Intensity Provider Class."""
```

Comment:
Put an empty line after the module docstring and add `from __future__ import annotations` as the first import. This applies to every file.